File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        def op(c: str):
            if (c=="+"):
                return True 
            if (c=="-"):
                return True
            if (c=="*"):
                return True
            if (c=="/"):
                return True
            return False
            

        n = len(tokens)
        if (n==1):
            return int(tokens[0])
        
        s = []
        for i in range(len(tokens)):
            c = tokens[i]
            if not op(c):
                s.append(int(c))
                continue

            ans = 0
            a1=s[-1]
            s.pop()
            a2=s[-1]

            s.pop()
            if (c=="+"):
                ans = a1+a2
            if (c=="-"):
                ans = a2-a1
            if (c=="*"):
                ans = a2*a1
            if (c=="/"):
                ans = (int)(a2/a1)
            s.append(ans)

        return s[-1]
            


            


        

File: Data Structures & Algorithms/validate-parentheses/submission-2.cpp
class Solution {
public:
    char op(char c){
        if(c==')') return '(';
        if(c=='}') return '{';
        if(c==']') return '[';
        return 'f';
    }
    bool isValid(string s) {
        int n = s.size();
        if (n%2!=0) return false;
        stack<int> st;
        for (const auto c : s){
            if (c=='(' || c=='{' || c=='['){
                st.push(c);
            }
            else{
                if(st.size()!=0 && st.top()==op(c)){
                    st.pop();
                    continue;
                }
                return false;
            }
        }
        if (st.size()==0)
        return true;
        return false;
    }
};
